darshan-util/pydarshan/darshan/cli/summary.py

```diff
@@ -244,21 +244,39 @@ def get_module_table(self):
             "Runtime Library Version": [job_data["metadata"]["lib_ver"], ""],
             "Log Format Version": [job_data["log_ver"], ""],
         }
-
         for mod in self.report.modules:
             # retrieve the module version and buffer sizes
             mod_version = self.report.modules[mod]["ver"]
-            # retrieve the buffer size converted to KiB
-            mod_buf_size = self.report.modules[mod]["len"] / 1024
+            # retrieve all the record and convert to pandas df
+            self.report.mod_read_all_records(mod)
+            df = self.report.records[mod].to_df()["counters"]
```

Comment:
One problem you'll have here, and it is reflected in some of the test failures, is that the result of `to_df()` can also be a list of DXT records, which are formatted quite differently, and you can't treat that as a simple dictionary.

Iterating through the full set of DXT data may also be time consuming, since it can be a huge Python list of records.

Reply:
I'm not sure it's super helpful to display much information on the DXT modules, beyond what the version is and whether the module has partial data. Unless there's a cheap way to calculate the total number of DXT file records?

If it's too expensive to calculate that, I'd say let's just avoid it altogether. The DXT modules are ultimately controlled by the POSIX and MPI-IO modules at runtime (i.e., DXT will likely have data on the same exact files as POSIX and MPI-IO modules), so there's not really a need for lots of DXT details since you can just infer that info from POSIX/MPI-IO modules.